src/sxmlc.c

```diff
@@ -1877,11 +1877,20 @@ int XMLDoc_parse_file_SAX(const SXML_CHAR* filename, const SAX_Callbacks* sax, v
 
 int XMLDoc_parse_buffer_SAX_len(const SXML_CHAR* buffer, int buffer_len, const SXML_CHAR* name, const SAX_Callbacks* sax, void* user)
 {
-	DataSourceBuffer dsb = { buffer, buffer_len, 0 };
-#ifdef _MSC_VER
-	SAX_Data sd = {0};
-#else
+#ifdef __CODEGEARC__
 	SAX_Data sd;
+	DataSourceBuffer dsb;
+	dsb.buf = buffer;
+	dsb.buf_len = buffer_len;
+	dsb.cur_pos = 0;
+#else
+	DataSourceBuffer dsb = { buffer, buffer_len, 0 };
+
+	#ifdef _MSC_VER

```

Comment:
I understand this is specific to MS Visual Studio. Does it complain about a structure not being initialized?

Reply:
The reason why I added this ifdef (as well as some similar in several other locations) is that I have a huge list of warnings claimed by Visual Studio otherwise, saying more or less that this value may be used without be initialized. It was somewhat annoying for me, because this drowned out other important warnings I should not miss.